pkg/render/intrusion_detection.go

```diff
@@ -97,15 +103,20 @@ func IntrusionDetection(cfg *IntrusionDetectionConfiguration) Component {
 
 // IntrusionDetectionConfiguration contains all the config information needed to render the component.
 type IntrusionDetectionConfiguration struct {
-	LogCollector          *operatorv1.LogCollector
-	ESSecrets             []*corev1.Secret
-	Installation          *operatorv1.InstallationSpec
-	ESClusterConfig       *relasticsearch.ClusterConfig
-	PullSecrets           []*corev1.Secret
-	Openshift             bool
-	ClusterDomain         string
-	ESLicenseType         ElasticsearchLicenseType
-	ManagedCluster        bool
+	IntrusionDetection operatorv1.IntrusionDetection
+	LogCollector       *operatorv1.LogCollector
+	ESSecrets          []*corev1.Secret
+	Installation       *operatorv1.InstallationSpec
+	ESClusterConfig    *relasticsearch.ClusterConfig
+	PullSecrets        []*corev1.Secret
+	Openshift          bool
+	ClusterDomain      string
+	ESLicenseType      ElasticsearchLicenseType
+	ManagedCluster     bool
+
+	// PVC fields Spec fields are immutable, set to true when and existing AD PVC
+	// is not found as to avoind
```

Comment:
typo: avoind

Also looks like the comment isn't 100% complete :grin: 

Also, we could probably avoid the need to have this flag altogether by using the `mergeState` logic to handle this.

Since this is common logic that, IIUC, would apply to ALL PVCs that we create, we could have `mergeState` check if the object already exists, and skip it if so.

https://github.com/tigera/operator/blob/a1a58b7d6a29786602fe6bd6c881ec4d52ee6826/pkg/controller/utils/component.go#L265-L266

Or, we could even do something sneaky like what we do for Jobs and delete / recreate the PVC, but that might not work because PVCs won't actually be removed until the pods using them are torn down as well. 

^ I don't think we need to block this PR on that, but might be something to look into as a follow on.

Reply:
added a TODO for this https://github.com/tigera/operator/pull/2094/commits/0242a6e687adf0bac93f81213982ec3928c81dd8

Reply:
Would rather not leave TODOs in the code. If we want to track something, we should raise it as a github issue so that it doesn't get lost.

Reply:
removed https://github.com/tigera/operator/pull/2094/commits/fe81c4cb19794ca1b97dd65394e9276d1e3a623d will create a github issue